parser/src/main/java/com/linkedin/avroutil1/util/ConfigurableAvroSchemaComparator.java

```diff
@@ -39,13 +42,27 @@ public class ConfigurableAvroSchemaComparator {
   public static boolean equals(AvroSchema a, AvroSchema b, SchemaComparisonConfiguration config) {
     validateConfig(config);
     Set<SeenPair> seen = new HashSet<>(3);
+    List<AvroSchemaDifference> differences = new ArrayList<>();
     try {
-      return equals(a, b, config, seen);
+      equals(a, b, config, seen, differences);
+      return differences.isEmpty();
     } catch (IOException e) {
       return false;
     }
   }
 
+  public static List<AvroSchemaDifference> findDifference(AvroSchema a, AvroSchema b, SchemaComparisonConfiguration config) {
+    validateConfig(config);
+    Set<SeenPair> seen = new HashSet<>(3);
+    List<AvroSchemaDifference> differences = new ArrayList<>();
+    try {
+      equals(a, b, config, seen, differences);
+    } catch (IOException ignored) {
+      // ignored
+    }
+    return differences;
+  }
```

Comment:
you added a new public method **findDifference**, how do you expect users call these two public methods, **equals** and  **findDifference**?
if users call call **equals** first, if returns false, then call **findDifference** to get the difference? this is low efficient, since they compare the schemas twice.

Users call **findDifference** standalone?  the problem is that the exception is not handled in this method. So if IOException is thrown during the method call, this method ignore the exception, the difference returned might be empty, so users get empty differences, and they think two schemas are identical.

Also add Java doc for all public methods.

Reply:
I think that `findDifference` and `equals` serve 2 different use cases. `equals` is just if you need to know if the schemas are equivalent, `findDifference` is for user-facing tools to find differences.

Reply:
@dg-builder @xiaoyu-yang-gh 
1. Instead of writing separate methods having similar code, I have introduced `fastFail` flag. This flag can be set to `true` for equals and it will fail on the first mismatch. We can set it to `false` for the `findDifference` and the code will go on to find all the differences in the schemas. Plus, this will improve code reusability and optimize performance. 
2. Throwing a `IllegalStateException` for any `IOExceptions`